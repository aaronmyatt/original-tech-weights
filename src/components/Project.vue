<template>
<ul class="project-container list-group">
  <li class="project-name card-title list-group-item d-flex w-100 justify-content-between">
    <h5>{{ projectName }}</h5>
  </li>

  <li class="project-description list-group-item d-flex w-100 justify-content-between">
    {{ projectDescription }}
  </li>

  <li class="project-date list-group-item d-flex w-100 justify-content-between">
    {{ projectDate }}
  </li>

  <li class="project-owner list-group-item d-flex w-100 justify-content-between">
    {{ projectOwner }}
  </li>

  <li class="project-repo-url list-group-item d-flex w-100 justify-content-between">
    {{ projectUrl }}
  </li>

  <li class="project-tech-weights list-group-item">
    <WeightVisualiser :weightList="getWeightList()"/>
  </li>
</ul>
</template>

<script>
import WeightVisualiser from './WeightVisualiser';

export default {
  name: 'Project',
  components: {
    WeightVisualiser,
  },
  props: [
    'projectName',
    'projectDescription',
    'projectDate',
    'projectOwner',
    'projectUrl',
    'projectWeights',
  ],
  methods: {
    getWeightList() {
      return this.projectWeights;
    },
  },
};
</script>

<style scoped>
</style>
