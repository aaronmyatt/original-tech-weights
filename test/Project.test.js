import { shallow } from 'vue-test-utils'
import Project from '../src/components/Project'

describe('Project', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Project, {
        propsData: {
            projectName: 'Test project',
            projectDescription: 'Test description',
            projectDate: new Date().toLocaleDateString(),
            projectOwner: 'Self',
            projectUrl: 'github.com/test_project'
        },
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Props', () => {

      it('renders the project name', () => {
          expect(cmp.find('.project-container .project-name').text().trim()).toBe(cmp.vm.projectName)
      })

      it('renders the project description', () => {
          expect(cmp.find('.project-container .project-description').text().trim()).toBe(cmp.vm.projectDescription)
      })

      it('renders the project date', () => {
          expect(cmp.find('.project-container .project-date').text().trim()).toBe(cmp.vm.projectDate)
      })

      it('renders the project owner', () => {
          expect(cmp.find('.project-container .project-owner').text().trim()).toBe(cmp.vm.projectOwner)
      })

      it('renders the project repo url', () => {
          expect(cmp.find('.project-container .project-repo-url').html()).toContain(cmp.vm.projectUrl)
      })

  })

})
