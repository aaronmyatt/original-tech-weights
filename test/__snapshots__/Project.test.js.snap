// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project has the expected html structure 1`] = `
<ul
  class="project-container list-group"
>
  <li
    class="project-name card-title list-group-item d-flex w-100 justify-content-between"
  >
    <h5>
      Test project
    </h5>
  </li>
   
  <li
    class="project-description list-group-item d-flex w-100 justify-content-between"
  >
    
    Test description
  
  </li>
   
  <li
    class="project-date list-group-item d-flex w-100 justify-content-between"
  >
    
    2017-10-25
  
  </li>
   
  <li
    class="project-owner list-group-item d-flex w-100 justify-content-between"
  >
    
    Self
  
  </li>
   
  <li
    class="project-repo-url list-group-item d-flex w-100 justify-content-between"
  >
    
    github.com/test_project
  
  </li>
   
  <li
    class="project-tech-weights list-group-item"
  >
    <!---->
  </li>
</ul>
`;
